dali/kernels/imgproc/pointwise/linear_transformation_gpu_test.cu

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+#include "dali/core/geom/mat.h"
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/tensor_shape.h"
+#include "dali/kernels/common/copy.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+#include "dali/kernels/imgproc/pointwise/linear_transformation_gpu.h"
+#include "dali/kernels/imgproc/color_manipulation/color_manipulation_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace test {
+
+
+namespace {
+
+/**
+ * Rounding to nearest even (like GPU does it)
+ */
+template <class Out>
+std::enable_if_t<std::is_integral<Out>::value, Out> custom_round(float val) {
+  return ConvertSat<Out>(std::nearbyint(val));
```

Comment:
Why do you insist on this? BTW - are you even sure that the rounding mode is indeed FE_TONEAREST?

Reply:
Done